docs/python-api.rst

```diff
@@ -1,34 +1,91 @@
 Python API
 ==========
```

Comment:
See comment above. Can we get rid of this file, and merge this documentation into the submodules in the appropriate place, e.g. `Installation` into `CONTRIBUTING.md`, the stuff about loss functions into `training.rst` etc?